makefile

```diff
@@ -1,27 +1,56 @@
-# trying this out under-the-radar as a framework-agnostic
-# build/install etc commands for frontend.
-# feel free to use it, but it may change/disappear
+default: help
 
-default:
-	grunt compile
+watch: compile-dev
+	@cd dev && make watch
 
-clean:
-	rm -rf node_modules
+compile:
+	@grunt compile-assets
 
-install:
-	npm prune && npm install
-	cd dev && make install
+compile-dev:
+	@grunt compile-assets --dev
 
-reinstall:
-	$(MAKE) clean
-	$(MAKE) install
+install: install-application install-dev
+	@node dev/message.js install
+
+reinstall: uninstall install
+
+uninstall: uninstall-application uninstall-dev
+	@echo 'All 3rd party dependencies have been uninstalled.'
 
 test:
-	grunt test --dev
+	@grunt test --dev
 
 validate:
-	grunt validate
+	@grunt validate
```

Comment:
Would we want to be able to validate css and js separately via make like running ``grunt validate:css``? Same with compile? Not sure if it's needed at the higher level (or whether the watcher on compile compiles specific files?)